x/concentrated-liquidity/position.go

```diff
@@ -81,26 +78,26 @@ func (k Keeper) hasFullPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAdd
 	return store.Has(key)
 }
 
-// GetPosition checks if a position exists at the provided upper and lower ticks and freezeDuration time for the given owner. Returns position if found.
-func (k Keeper) GetPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick, upperTick int64, joinTime time.Time, freezeDuration time.Duration) (*model.Position, error) {
+// GetPositionLiquidity checks if a position exists at the provided upper and lower ticks and freezeDuration time for the given owner. Returns position if found.
+func (k Keeper) GetPositionLiquidity(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick, upperTick int64, joinTime time.Time, freezeDuration time.Duration) (sdk.Dec, error) {
 	store := ctx.KVStore(k.storeKey)
-	positionStruct := &model.Position{}
+	liquidityStruct := &sdk.DecProto{}
 	key := types.KeyFullPosition(poolId, owner, lowerTick, upperTick, joinTime, freezeDuration)
 
-	found, err := osmoutils.Get(store, key, positionStruct)
+	found, err := osmoutils.Get(store, key, liquidityStruct)
```

Comment:
Not sure if it matters, but I thinks it should be:

liquidityStruct := sdk.DecProto{} on line 84

then we call &liquidityStruct on line 87

Since we are no longer returning a dereferenced model.Position and instead are returning a normal sdk.Dec.

It probably works the same, but just caught my eye

Reply:
This lgtm. On lin 87, we are providing a pointer because we want the `osmoutils.Get` to populate the `liquidityStruct` by mutating the pointer. Otherwise, there is no way to use a helper getter to get the `sdk.Dec` from store. That is, the object would not be mutated if we were to pass it in by value to `osmoutils.Get(...)`

Let me know if that addressed your concern